src/engine/strat_engine/blockdev.rs

```diff
@@ -55,10 +56,67 @@ pub fn resolve_devices(paths: &[&Path]) -> io::Result<BTreeSet<Device>> {
     Ok(devices)
 }
 
+#[derive(Debug)]
+pub struct BlockDevs {
+    pub devs: BTreeMap<DevUuid, BlockDev>,
+}
+
+impl BlockDevs {
+    pub fn new(devs: BTreeMap<DevUuid, BlockDev>) -> BlockDevs {
+        BlockDevs { devs: devs }
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.devs.is_empty()
+    }
+
+    /// Read the latest json across all blockdevs
```

Comment:
"json" is the format of the data you're reading, not its meaning or significance.